{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ensembl import Ensembl\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'A': {'spdi': ['NC_000022.11:37073552:C:T', 'LRG_1128:41010:G:A'],\n",
       "   'input': 'NP_001361433.1:p.Asp512Asn',\n",
       "   'hgvsc': ['ENST00000346753.9:c.1534G>A',\n",
       "    'ENST00000381792.6:c.1534G>A',\n",
       "    'ENST00000406725.6:c.1534G>A',\n",
       "    'ENST00000406856.7:c.1534G>A',\n",
       "    'ENST00000676104.1:c.1534G>A',\n",
       "    'NM_001289000.2:c.1534G>A',\n",
       "    'NM_001289001.2:c.1534G>A',\n",
       "    'NM_001374504.1:c.1534G>A',\n",
       "    'NM_153609.4:c.1534G>A',\n",
       "    'XM_011529989.3:c.1102G>A',\n",
       "    'XM_024452167.2:c.1534G>A',\n",
       "    'XM_024452168.2:c.1534G>A',\n",
       "    'XM_047441170.1:c.1102G>A',\n",
       "    'XM_047441171.1:c.1102G>A',\n",
       "    'LRG_1128t1:c.1534G>A',\n",
       "    'LRG_1128t2:c.1534G>A'],\n",
       "   'id': ['rs137853120', 'COSV60981142', 'CM081842'],\n",
       "   'hgvsg': ['NC_000022.11:g.37073553C>T', 'LRG_1128:g.41011G>A'],\n",
       "   'hgvsp': ['ENSP00000334962.6:p.Asp512Asn',\n",
       "    'ENSP00000371211.2:p.Asp512Asn',\n",
       "    'ENSP00000385453.1:p.Asp512Asn',\n",
       "    'ENSP00000384964.1:p.Asp512Asn',\n",
       "    'ENSP00000501573.1:p.Asp512Asn',\n",
       "    'NP_001275929.1:p.Asp512Asn',\n",
       "    'NP_001275930.1:p.Asp512Asn',\n",
       "    'NP_001361433.1:p.Asp512Asn',\n",
       "    'NP_705837.2:p.Asp512Asn',\n",
       "    'XP_011528291.1:p.Asp368Asn',\n",
       "    'XP_024307935.1:p.Asp512Asn',\n",
       "    'XP_024307936.1:p.Asp512Asn',\n",
       "    'XP_047297126.1:p.Asp368Asn',\n",
       "    'XP_047297127.1:p.Asp368Asn',\n",
       "    'LRG_1128p1:p.Asp512Asn',\n",
       "    'LRG_1128p2:p.Asp512Asn']}}]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl = Ensembl()\n",
    "ensembl.variant_recoder(\"NP_001361433.1:p.Asp512Asn\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'T': {'input': 'rs137853120',\n",
       "   'spdi': ['NC_000022.11:37073552:C:T', 'LRG_1128:41010:G:A'],\n",
       "   'id': ['rs137853120', 'COSV60981142', 'CM081842'],\n",
       "   'hgvsg': ['NC_000022.11:g.37073553C>T', 'LRG_1128:g.41011G>A'],\n",
       "   'hgvsp': ['ENSP00000334962.6:p.Asp512Asn',\n",
       "    'ENSP00000371211.2:p.Asp512Asn',\n",
       "    'ENSP00000385453.1:p.Asp512Asn',\n",
       "    'ENSP00000384964.1:p.Asp512Asn',\n",
       "    'ENSP00000501573.1:p.Asp512Asn',\n",
       "    'NP_001275929.1:p.Asp512Asn',\n",
       "    'NP_001275930.1:p.Asp512Asn',\n",
       "    'NP_001361433.1:p.Asp512Asn',\n",
       "    'NP_705837.2:p.Asp512Asn',\n",
       "    'XP_011528291.1:p.Asp368Asn',\n",
       "    'XP_024307935.1:p.Asp512Asn',\n",
       "    'XP_024307936.1:p.Asp512Asn',\n",
       "    'XP_047297126.1:p.Asp368Asn',\n",
       "    'XP_047297127.1:p.Asp368Asn',\n",
       "    'LRG_1128p1:p.Asp512Asn',\n",
       "    'LRG_1128p2:p.Asp512Asn'],\n",
       "   'hgvsc': ['ENST00000346753.9:c.1534G>A',\n",
       "    'ENST00000381792.6:c.1534G>A',\n",
       "    'ENST00000406725.6:c.1534G>A',\n",
       "    'ENST00000406856.7:c.1534G>A',\n",
       "    'ENST00000676104.1:c.1534G>A',\n",
       "    'NM_001289000.2:c.1534G>A',\n",
       "    'NM_001289001.2:c.1534G>A',\n",
       "    'NM_001374504.1:c.1534G>A',\n",
       "    'NM_153609.4:c.1534G>A',\n",
       "    'XM_011529989.3:c.1102G>A',\n",
       "    'XM_024452167.2:c.1534G>A',\n",
       "    'XM_024452168.2:c.1534G>A',\n",
       "    'XM_047441170.1:c.1102G>A',\n",
       "    'XM_047441171.1:c.1102G>A',\n",
       "    'LRG_1128t1:c.1534G>A',\n",
       "    'LRG_1128t2:c.1534G>A']}},\n",
       " {'T': {'id': ['rs137853119', 'CM081843'],\n",
       "   'input': 'rs137853119',\n",
       "   'spdi': ['NC_000022.11:37075179:C:T', 'LRG_1128:39383:G:A'],\n",
       "   'hgvsc': ['ENST00000346753.9:c.1297G>A',\n",
       "    'ENST00000381792.6:c.1297G>A',\n",
       "    'ENST00000406725.6:c.1297G>A',\n",
       "    'ENST00000406856.7:c.1297G>A',\n",
       "    'ENST00000676104.1:c.1297G>A',\n",
       "    'NM_001289000.2:c.1297G>A',\n",
       "    'NM_001289001.2:c.1297G>A',\n",
       "    'NM_001374504.1:c.1297G>A',\n",
       "    'NM_153609.4:c.1297G>A',\n",
       "    'XM_011529989.3:c.865G>A',\n",
       "    'XM_024452167.2:c.1297G>A',\n",
       "    'XM_024452168.2:c.1297G>A',\n",
       "    'XM_047441170.1:c.865G>A',\n",
       "    'XM_047441171.1:c.865G>A',\n",
       "    'LRG_1128t1:c.1297G>A',\n",
       "    'LRG_1128t2:c.1297G>A'],\n",
       "   'hgvsg': ['NC_000022.11:g.37075180C>T', 'LRG_1128:g.39384G>A'],\n",
       "   'hgvsp': ['ENSP00000334962.6:p.Gly433Arg',\n",
       "    'ENSP00000371211.2:p.Gly433Arg',\n",
       "    'ENSP00000385453.1:p.Gly433Arg',\n",
       "    'ENSP00000384964.1:p.Gly433Arg',\n",
       "    'ENSP00000501573.1:p.Gly433Arg',\n",
       "    'NP_001275929.1:p.Gly433Arg',\n",
       "    'NP_001275930.1:p.Gly433Arg',\n",
       "    'NP_001361433.1:p.Gly433Arg',\n",
       "    'NP_705837.2:p.Gly433Arg',\n",
       "    'XP_011528291.1:p.Gly289Arg',\n",
       "    'XP_024307935.1:p.Gly433Arg',\n",
       "    'XP_024307936.1:p.Gly433Arg',\n",
       "    'XP_047297126.1:p.Gly289Arg',\n",
       "    'XP_047297127.1:p.Gly289Arg',\n",
       "    'LRG_1128p1:p.Gly433Arg',\n",
       "    'LRG_1128p2:p.Gly433Arg']}}]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl.variant_recoder(['rs137853120', 'rs137853119'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'source': 'Variants (including SNPs and indels) imported from dbSNP',\n",
       " 'most_severe_consequence': 'missense_variant',\n",
       " 'name': 'rs137853120',\n",
       " 'var_class': 'SNP',\n",
       " 'mappings': [{'strand': 1,\n",
       "   'start': 37073553,\n",
       "   'seq_region_name': '22',\n",
       "   'ancestral_allele': 'C',\n",
       "   'coord_system': 'chromosome',\n",
       "   'assembly_name': 'GRCh38',\n",
       "   'location': '22:37073553-37073553',\n",
       "   'allele_string': 'C/T',\n",
       "   'end': 37073553}],\n",
       " 'synonyms': ['NM_153609.3:c.1561G>A',\n",
       "  'NM_001289000.2:c.1534G>A',\n",
       "  'NM_153609.2:c.1561G>A',\n",
       "  'NP_001361433.1:p.Asp512Asn',\n",
       "  'NM_001374504.1:c.1534G>A',\n",
       "  'NM_001289001.2:c.1534G>A',\n",
       "  'NM_153609.4:c.1534G>A',\n",
       "  'NM_001289000.1:c.1534G>A',\n",
       "  'NP_001275930.1:p.Asp512Asn',\n",
       "  'NP_001275929.1:p.Asp512Asn',\n",
       "  'NM_001289001.1:c.1534G>A',\n",
       "  'NP_705837.2:p.Asp512Asn',\n",
       "  'VAR_044436',\n",
       "  'VCV000001404',\n",
       "  'RCV001375845',\n",
       "  'RCV000001469',\n",
       "  '609862.0006'],\n",
       " 'clinical_significance': ['likely pathogenic', 'pathogenic'],\n",
       " 'evidence': ['Frequency',\n",
       "  '1000Genomes',\n",
       "  'Cited',\n",
       "  'ESP',\n",
       "  'Phenotype_or_Disease',\n",
       "  'ExAC',\n",
       "  'TOPMed',\n",
       "  'gnomAD'],\n",
       " 'MAF': 0.0002,\n",
       " 'ambiguity': 'Y',\n",
       " 'minor_allele': 'T'}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl.variation('rs137853120')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rs137853120</th>\n",
       "      <th>rs137853119</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>minor_allele</th>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mappings</th>\n",
       "      <td>[{'start': 37073553, 'allele_string': 'C/T', '...</td>\n",
       "      <td>[{'location': '22:37075180-37075180', 'strand'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <td>rs137853120</td>\n",
       "      <td>rs137853119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>clinical_significance</th>\n",
       "      <td>[likely pathogenic, pathogenic]</td>\n",
       "      <td>[pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>synonyms</th>\n",
       "      <td>[NM_153609.2:c.1561G&gt;A, NP_001275930.1:p.Asp51...</td>\n",
       "      <td>[RCV001851548, VCV000001403, RCV000001468, NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ambiguity</th>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>evidence</th>\n",
       "      <td>[Frequency, 1000Genomes, Cited, ESP, Phenotype...</td>\n",
       "      <td>[Frequency, Cited, Phenotype_or_Disease, TOPMe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF</th>\n",
       "      <td>0.0002</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>source</th>\n",
       "      <td>Variants (including SNPs and indels) imported ...</td>\n",
       "      <td>Variants (including SNPs and indels) imported ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>var_class</th>\n",
       "      <td>SNP</td>\n",
       "      <td>SNP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               rs137853120  \\\n",
       "minor_allele                                                             T   \n",
       "mappings                 [{'start': 37073553, 'allele_string': 'C/T', '...   \n",
       "most_severe_consequence                                   missense_variant   \n",
       "name                                                           rs137853120   \n",
       "clinical_significance                      [likely pathogenic, pathogenic]   \n",
       "synonyms                 [NM_153609.2:c.1561G>A, NP_001275930.1:p.Asp51...   \n",
       "ambiguity                                                                Y   \n",
       "evidence                 [Frequency, 1000Genomes, Cited, ESP, Phenotype...   \n",
       "MAF                                                                 0.0002   \n",
       "source                   Variants (including SNPs and indels) imported ...   \n",
       "var_class                                                              SNP   \n",
       "\n",
       "                                                               rs137853119  \n",
       "minor_allele                                                          None  \n",
       "mappings                 [{'location': '22:37075180-37075180', 'strand'...  \n",
       "most_severe_consequence                                   missense_variant  \n",
       "name                                                           rs137853119  \n",
       "clinical_significance                                         [pathogenic]  \n",
       "synonyms                 [RCV001851548, VCV000001403, RCV000001468, NM_...  \n",
       "ambiguity                                                                Y  \n",
       "evidence                 [Frequency, Cited, Phenotype_or_Disease, TOPMe...  \n",
       "MAF                                                                   None  \n",
       "source                   Variants (including SNPs and indels) imported ...  \n",
       "var_class                                                              SNP  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(ensembl.variation(['rs137853120', 'rs137853119']))\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_consequences</th>\n",
       "      <th>colocated_variants</th>\n",
       "      <th>allele_string</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>input</th>\n",
       "      <th>start</th>\n",
       "      <th>seq_region_name</th>\n",
       "      <th>id</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[{'polyphen_prediction': 'probably_damaging', ...</td>\n",
       "      <td>[{'start': 37073553, 'strand': 1, 'seq_region_...</td>\n",
       "      <td>C/T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>rs137853120</td>\n",
       "      <td>37073553</td>\n",
       "      <td>22</td>\n",
       "      <td>rs137853120</td>\n",
       "      <td>37073553</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             transcript_consequences  \\\n",
       "0  [{'polyphen_prediction': 'probably_damaging', ...   \n",
       "\n",
       "                                  colocated_variants allele_string  \\\n",
       "0  [{'start': 37073553, 'strand': 1, 'seq_region_...           C/T   \n",
       "\n",
       "  most_severe_consequence assembly_name        input     start  \\\n",
       "0        missense_variant        GRCh38  rs137853120  37073553   \n",
       "\n",
       "  seq_region_name           id       end  strand  \n",
       "0              22  rs137853120  37073553       1  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(ensembl.vep_id('rs137853120'))\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>transcript_consequences</th>\n",
       "      <td>[{'polyphen_prediction': 'probably_damaging', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colocated_variants</th>\n",
       "      <td>[{'start': 37073553, 'strand': 1, 'seq_region_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allele_string</th>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_name</th>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <td>rs137853120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>start</th>\n",
       "      <td>37073553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seq_region_name</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <td>rs137853120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>end</th>\n",
       "      <td>37073553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strand</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         0\n",
       "transcript_consequences  [{'polyphen_prediction': 'probably_damaging', ...\n",
       "colocated_variants       [{'start': 37073553, 'strand': 1, 'seq_region_...\n",
       "allele_string                                                          C/T\n",
       "most_severe_consequence                                   missense_variant\n",
       "assembly_name                                                       GRCh38\n",
       "input                                                          rs137853120\n",
       "start                                                             37073553\n",
       "seq_region_name                                                         22\n",
       "id                                                             rs137853120\n",
       "end                                                               37073553\n",
       "strand                                                                   1"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strand</th>\n",
       "      <th>id</th>\n",
       "      <th>colocated_variants</th>\n",
       "      <th>start</th>\n",
       "      <th>allele_string</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>end</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>seq_region_name</th>\n",
       "      <th>transcript_consequences</th>\n",
       "      <th>input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs137853120</td>\n",
       "      <td>[{'id': 'CM081842', 'start': 37073553, 'strand...</td>\n",
       "      <td>37073553</td>\n",
       "      <td>C/T</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>37073553</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>22</td>\n",
       "      <td>[{'biotype': 'protein_coding', 'hgnc_id': 'HGN...</td>\n",
       "      <td>rs137853120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs137853119</td>\n",
       "      <td>[{'allele_string': 'HGMD_MUTATION', 'start': 3...</td>\n",
       "      <td>37075180</td>\n",
       "      <td>C/T</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>37075180</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>22</td>\n",
       "      <td>[{'codons': 'Ggg/Agg', 'sift_prediction': 'del...</td>\n",
       "      <td>rs137853119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   strand           id                                 colocated_variants  \\\n",
       "0       1  rs137853120  [{'id': 'CM081842', 'start': 37073553, 'strand...   \n",
       "1       1  rs137853119  [{'allele_string': 'HGMD_MUTATION', 'start': 3...   \n",
       "\n",
       "      start allele_string assembly_name       end most_severe_consequence  \\\n",
       "0  37073553           C/T        GRCh38  37073553        missense_variant   \n",
       "1  37075180           C/T        GRCh38  37075180        missense_variant   \n",
       "\n",
       "  seq_region_name                            transcript_consequences  \\\n",
       "0              22  [{'biotype': 'protein_coding', 'hgnc_id': 'HGN...   \n",
       "1              22  [{'codons': 'Ggg/Agg', 'sift_prediction': 'del...   \n",
       "\n",
       "         input  \n",
       "0  rs137853120  \n",
       "1  rs137853119  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(ensembl.vep_id(['rs137853120', 'rs137853119']))\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>strand</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <td>rs137853120</td>\n",
       "      <td>rs137853119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colocated_variants</th>\n",
       "      <td>[{'id': 'CM081842', 'start': 37073553, 'strand...</td>\n",
       "      <td>[{'allele_string': 'HGMD_MUTATION', 'start': 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>start</th>\n",
       "      <td>37073553</td>\n",
       "      <td>37075180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allele_string</th>\n",
       "      <td>C/T</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_name</th>\n",
       "      <td>GRCh38</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>end</th>\n",
       "      <td>37073553</td>\n",
       "      <td>37075180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seq_region_name</th>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_consequences</th>\n",
       "      <td>[{'biotype': 'protein_coding', 'hgnc_id': 'HGN...</td>\n",
       "      <td>[{'codons': 'Ggg/Agg', 'sift_prediction': 'del...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <td>rs137853120</td>\n",
       "      <td>rs137853119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         0  \\\n",
       "strand                                                                   1   \n",
       "id                                                             rs137853120   \n",
       "colocated_variants       [{'id': 'CM081842', 'start': 37073553, 'strand...   \n",
       "start                                                             37073553   \n",
       "allele_string                                                          C/T   \n",
       "assembly_name                                                       GRCh38   \n",
       "end                                                               37073553   \n",
       "most_severe_consequence                                   missense_variant   \n",
       "seq_region_name                                                         22   \n",
       "transcript_consequences  [{'biotype': 'protein_coding', 'hgnc_id': 'HGN...   \n",
       "input                                                          rs137853120   \n",
       "\n",
       "                                                                         1  \n",
       "strand                                                                   1  \n",
       "id                                                             rs137853119  \n",
       "colocated_variants       [{'allele_string': 'HGMD_MUTATION', 'start': 3...  \n",
       "start                                                             37075180  \n",
       "allele_string                                                          C/T  \n",
       "assembly_name                                                       GRCh38  \n",
       "end                                                               37075180  \n",
       "most_severe_consequence                                   missense_variant  \n",
       "seq_region_name                                                         22  \n",
       "transcript_consequences  [{'codons': 'Ggg/Agg', 'sift_prediction': 'del...  \n",
       "input                                                          rs137853119  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seq_region_name</th>\n",
       "      <th>allele_string</th>\n",
       "      <th>transcript_consequences</th>\n",
       "      <th>end</th>\n",
       "      <th>colocated_variants</th>\n",
       "      <th>strand</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>start</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "      <td>22</td>\n",
       "      <td>G/A</td>\n",
       "      <td>[{'cdna_end': 1675, 'codons': 'Gac/Aac', 'sift...</td>\n",
       "      <td>37073553</td>\n",
       "      <td>[{'start': 37073553, 'id': 'CM081842', 'allele...</td>\n",
       "      <td>-1</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>37073553</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           id seq_region_name allele_string  \\\n",
       "0  NP_001361433.1:p.Asp512Asn              22           G/A   \n",
       "\n",
       "                             transcript_consequences       end  \\\n",
       "0  [{'cdna_end': 1675, 'codons': 'Gac/Aac', 'sift...  37073553   \n",
       "\n",
       "                                  colocated_variants  strand assembly_name  \\\n",
       "0  [{'start': 37073553, 'id': 'CM081842', 'allele...      -1        GRCh38   \n",
       "\n",
       "      start most_severe_consequence                       input  \n",
       "0  37073553        missense_variant  NP_001361433.1:p.Asp512Asn  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(ensembl.vep_hgvs(\"NP_001361433.1:p.Asp512Asn\"))\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seq_region_name</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allele_string</th>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_consequences</th>\n",
       "      <td>[{'cdna_end': 1675, 'codons': 'Gac/Aac', 'sift...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>end</th>\n",
       "      <td>37073553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colocated_variants</th>\n",
       "      <td>[{'start': 37073553, 'id': 'CM081842', 'allele...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strand</th>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_name</th>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>start</th>\n",
       "      <td>37073553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         0\n",
       "id                                              NP_001361433.1:p.Asp512Asn\n",
       "seq_region_name                                                         22\n",
       "allele_string                                                          G/A\n",
       "transcript_consequences  [{'cdna_end': 1675, 'codons': 'Gac/Aac', 'sift...\n",
       "end                                                               37073553\n",
       "colocated_variants       [{'start': 37073553, 'id': 'CM081842', 'allele...\n",
       "strand                                                                  -1\n",
       "assembly_name                                                       GRCh38\n",
       "start                                                             37073553\n",
       "most_severe_consequence                                   missense_variant\n",
       "input                                           NP_001361433.1:p.Asp512Asn"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_region_name</th>\n",
       "      <th>end</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <th>transcript_consequences</th>\n",
       "      <th>input</th>\n",
       "      <th>colocated_variants</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>id</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>allele_string</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>22</td>\n",
       "      <td>37073553</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[{'consequence_terms': ['missense_variant'], '...</td>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "      <td>[{'seq_region_name': '22', 'phenotype_or_disea...</td>\n",
       "      <td>37073553</td>\n",
       "      <td>-1</td>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22</td>\n",
       "      <td>37098426</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[{'gene_id': 'ENSG00000187045', 'gene_symbol_s...</td>\n",
       "      <td>NM_001374504.1:c.326C&gt;A</td>\n",
       "      <td>[{'seq_region_name': '22', 'clin_sig_allele': ...</td>\n",
       "      <td>37098426</td>\n",
       "      <td>-1</td>\n",
       "      <td>NM_001374504.1:c.326C&gt;A</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>C/A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seq_region_name       end most_severe_consequence  \\\n",
       "0              22  37073553        missense_variant   \n",
       "1              22  37098426        missense_variant   \n",
       "\n",
       "                             transcript_consequences  \\\n",
       "0  [{'consequence_terms': ['missense_variant'], '...   \n",
       "1  [{'gene_id': 'ENSG00000187045', 'gene_symbol_s...   \n",
       "\n",
       "                        input  \\\n",
       "0  NP_001361433.1:p.Asp512Asn   \n",
       "1     NM_001374504.1:c.326C>A   \n",
       "\n",
       "                                  colocated_variants     start  strand  \\\n",
       "0  [{'seq_region_name': '22', 'phenotype_or_disea...  37073553      -1   \n",
       "1  [{'seq_region_name': '22', 'clin_sig_allele': ...  37098426      -1   \n",
       "\n",
       "                           id assembly_name allele_string  \n",
       "0  NP_001361433.1:p.Asp512Asn        GRCh38           G/A  \n",
       "1     NM_001374504.1:c.326C>A        GRCh38           C/A  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(ensembl.vep_hgvs(['NP_001361433.1:p.Asp512Asn', \"NM_001374504.1:c.326C>A\"]))\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>seq_region_name</th>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>end</th>\n",
       "      <td>37073553</td>\n",
       "      <td>37098426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>most_severe_consequence</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_consequences</th>\n",
       "      <td>[{'consequence_terms': ['missense_variant'], '...</td>\n",
       "      <td>[{'gene_id': 'ENSG00000187045', 'gene_symbol_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "      <td>NM_001374504.1:c.326C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colocated_variants</th>\n",
       "      <td>[{'seq_region_name': '22', 'phenotype_or_disea...</td>\n",
       "      <td>[{'seq_region_name': '22', 'clin_sig_allele': ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>start</th>\n",
       "      <td>37073553</td>\n",
       "      <td>37098426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strand</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <td>NP_001361433.1:p.Asp512Asn</td>\n",
       "      <td>NM_001374504.1:c.326C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_name</th>\n",
       "      <td>GRCh38</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allele_string</th>\n",
       "      <td>G/A</td>\n",
       "      <td>C/A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         0  \\\n",
       "seq_region_name                                                         22   \n",
       "end                                                               37073553   \n",
       "most_severe_consequence                                   missense_variant   \n",
       "transcript_consequences  [{'consequence_terms': ['missense_variant'], '...   \n",
       "input                                           NP_001361433.1:p.Asp512Asn   \n",
       "colocated_variants       [{'seq_region_name': '22', 'phenotype_or_disea...   \n",
       "start                                                             37073553   \n",
       "strand                                                                  -1   \n",
       "id                                              NP_001361433.1:p.Asp512Asn   \n",
       "assembly_name                                                       GRCh38   \n",
       "allele_string                                                          G/A   \n",
       "\n",
       "                                                                         1  \n",
       "seq_region_name                                                         22  \n",
       "end                                                               37098426  \n",
       "most_severe_consequence                                   missense_variant  \n",
       "transcript_consequences  [{'gene_id': 'ENSG00000187045', 'gene_symbol_s...  \n",
       "input                                              NM_001374504.1:c.326C>A  \n",
       "colocated_variants       [{'seq_region_name': '22', 'clin_sig_allele': ...  \n",
       "start                                                             37098426  \n",
       "strand                                                                  -1  \n",
       "id                                                 NM_001374504.1:c.326C>A  \n",
       "assembly_name                                                       GRCh38  \n",
       "allele_string                                                          C/A  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.T"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
